nixos/modules/services/networking/netbird.nix

```
@@ -1,171 +1,336 @@
-{
-  config,
-  lib,
-  pkgs,
-  ...
+{ config
+, lib
+, pkgs
+, ...
 }:
 
 let
   inherit (lib)
-    attrNames
+    attrValues
+    escapeShellArgs
+    flatten
     getExe
     literalExpression
     maintainers
     mapAttrs'
+    mapAttrsToList
+    mkAliasOptionModule
     mkDefault
-    mkEnableOption
     mkIf
     mkMerge
     mkOption
+    mkOptionDefault
     mkPackageOption
     nameValuePair
     optional
+    optionals
+    versionAtLeast
     versionOlder
     ;
 
   inherit (lib.types)
     attrsOf
+    bool
+    enum
+    package
     port
     str
     submodule
     ;
 
-  kernel = config.boot.kernelPackages;
+  inherit (config.boot) kernelPackages;
+  inherit (config.boot.kernelPackages) kernel;
 
   cfg = config.services.netbird;
+
+  toClientList = fn: map fn (attrValues cfg.clients);
+  toClientAttrs = fn: mapAttrs' (_: fn) cfg.clients;
 in
 {
   meta.maintainers = with maintainers; [
     misuzu
     thubrecht
+    nazarewk
   ];
   meta.doc = ./netbird.md;
 
+  imports = [
+    (mkAliasOptionModule [ "services" "netbird" "tunnels" ] [ "services" "netbird" "clients" ])
```

```suggestion
    (mkRemovedOptionModule [ "services" "netbird" "tunnels" ] "The option `services.netbird.tunnels` has been renamed to `services.netbird.clients`")
```
I don't think that many people have already started to use the `tunnels` option, not much disturbance would happen with the removal